src/dotnet/commands/dotnet-cache/CacheCommand.cs

```diff
@@ -0,0 +1,138 @@
+        private bool PreserveIntermediateDir { get; set; }
+
+        public List<string> ExtraMSBuildArguments { get; set; }
+
+        private CacheCommand()
+        {
+        }
+
+        private void PrepareForExecute(CommandOption argument)
+        {
+            string targetProjectPath = string.Empty;
+
+            if (argument == null || !argument.Values.Any())
+            {
+                throw new InvalidOperationException(LocalizableStrings.SpecifyEntries);
+            }
+            try
+            {
+                if (string.IsNullOrEmpty(IntermediateDir))
+                {
+                    var tempDir = Path.GetTempPath();
+                    var folder = Path.GetRandomFileName();
+                    IntermediateDir = Path.Combine(tempDir, folder);
+                }
+                else
+                {
+                    IntermediateDir = Path.GetFullPath(IntermediateDir);
+                }
+
+                if (Directory.Exists(IntermediateDir))
+                {
+                    throw new GracefulException(string.Format(LocalizableStrings.IntermediateDirExists, IntermediateDir));
+                }
+
+                Directory.CreateDirectory(IntermediateDir);
+                var givenProject = argument.Values.FirstOrDefault(); //TODO ensure multiple entries can be processed
+                var projectFile = Path.GetFileName(givenProject);
+                targetProjectPath = Path.Combine(IntermediateDir, projectFile);
+
+                File.Copy(givenProject, targetProjectPath);
+
+                ProjectPath = targetProjectPath;
+            }
+            catch (Exception e)
+            {
+                throw new GracefulException(e.ToString());
+            }
+        }
+        private void CleanupAfterExecute()
+        {
+            if (!PreserveIntermediateDir)
+            {
+                try
+                {
+                    Directory.Delete(IntermediateDir, true);
+                }
+                catch (System.Exception e)
+                {
+                    throw new GracefulException(e.ToString());
+                }
+            }
+        }
+
+        public int Execute()
+        {
+            int result = 0;
+            List<string> msbuildArgs = new List<string>();
+            if (!string.IsNullOrEmpty(ProjectPath))
+            {
+                msbuildArgs.Add(ProjectPath);
+            }
+
+            msbuildArgs.Add("/t:ComposeCache");
+
+            if (!string.IsNullOrEmpty(Framework))
+            {
+                msbuildArgs.Add($"/p:TargetFramework={Framework}");
+            }
+
+            if (!string.IsNullOrEmpty(Runtime))
+            {
+                msbuildArgs.Add($"/p:RuntimeIdentifier={Runtime}");
+            }
+
+            if (!string.IsNullOrEmpty(OutputPath))
+            {
+                OutputPath = Path.GetFullPath(OutputPath);
+                msbuildArgs.Add($"/p:ComposeDir={OutputPath}");
```

Comment:
You did not want to confuse the user or developers? Because, for the user, you are exposing this in the CLI as output, right?